test/com/amazon/ion/CloneTest.java

```diff
@@ -15,30 +15,53 @@
 
 package com.amazon.ion;
 
-import static com.amazon.ion.impl._Private_Utils.newSymbolToken;
-
-import com.amazon.ion.system.SimpleCatalog;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.ExpectedException;
+import com.amazon.ion.impl._Private_IonSystem;
+import com.amazon.ion.junit.IonAssert;
+import com.amazon.ion.system.IonSystemBuilder;
+import org.junit.jupiter.api.Test;
 
+import static com.amazon.ion.impl._Private_Utils.newSymbolToken;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class CloneTest
-    extends IonTestCase
 {
-    @Rule
-    public ExpectedException thrown = ExpectedException.none();
 
+    public static final _Private_IonSystem PRIVATE_ION_SYSTEM = (_Private_IonSystem) IonSystemBuilder.standard().build();
+
+    private static <V extends IonValue> V assertSelfCloneable(V original)
+    {
+        V clone = (V)original.clone();
+        return assertEqualsAndNotSame(original, clone);
+    }
+
+    private static <V extends IonValue> V assertSystemCloneable(V original)
+    {
+        V clone = (V)(system().clone(original));
+        return assertEqualsAndNotSame(original, clone);
+    }
```

Comment:
I just realized that it's probably cleanest to implement `assert*Cloneable` in terms of something like this:
```java
    private static <V extends IonValue> V assertClone(Function<V,V> cloner, V original)
    {
        return assertEqualsAndNotSame(original, (V)cloner.apply(original));
    }
```

The separate assertions are probably not even necessary then, instead just something like this:
```java
    assertClone(system()::clone, original);
    assertClone(original::clone, original);
```

Then `assertClone` *becomes* `assertEqualsAndNotTheSame` because really how can you tell otherwise? Meaning `assertEqualsAndNotTheSame` and both `assert*Cloneable` methods get inlined.

I'm almost certainly neglecting some ugly wrinkle of syntax needed to work with generics here 🤷 .
Sounds like a problem for posterity.